### ISSUE TITLE
DRM tags

### ISSUE THREAD

**Issue by Protektor-Desura** [desura/Desurium#93]
The issue is some would like DRM notification and at what levels it is. Should we go with just "No DRM" and DRM or something more? If we go with something more my suggestion is "No DRM" and "DRM Key" and "DRM Auth". This is a server side issue and not a client side issue.

**Comment by Jookia** [comment]
Well, DRM is kind of an umbrella. Maybe we should just have a DRM bool, and let the uploaders manually write information on the DRM.

**Comment by ssokolow** [comment]
See issue #80 for what's already been said.

The gist of it is that @naryl wants a three-state system differentiating between open-source, closed-source but un-DRMed, and DRMed.

I've done a bit of thinking on it and I think five gradations would make more sense:
1. Open-source 
2. Closed-source
3. CD-key or other unique identifier must be provided but doesn't phone home
4. Phones home once to validate key
5. Phones home every time you play

**Comment by Jookia** [comment]
I don't use closed-source software unless I have to, but I feel that we're intruding a little by marking things as closed/open source.

What about multiplayer login DRM?

**Comment by ssokolow** [comment]
Good point. I actually just recently contacted GOG.com about the meaning of "this game requires multiplayer keys" so I'm surprised I didn't think about it.

I suppose it could be re-structured like this instead:
- Single-player status
  1. Unfettered (open- or closed-source)
  2. CD-key but doesn't phone home
  3. Phones home once
  4. Phones home every time
- Multi-player status
  1. Open
  2. Internet play requires CDkey uniqueness, LAN play unrestricted
  3. LAN play requires CDkey uniqueness or is not implemented separately from online play

(I'm not sure how one would file the old Warcraft "up to 3 players per CD" approach. More to think on.)

...though I do still think it'd be nice to have at least a boolean for "source available". If for no other reason, because it'd save me a lot of hassle checking for every non-paid game I download. (Among other reasons, I like to know what is and isn't impossible to get running on ARM machines like the OpenPandora and the Raspberry Pi)

**Comment by Jookia** [comment]
Blockland has a weird way of DRM: Key required to build over 150 bricks in singleplayer and to play multiplayer, two people from the same IP can share a key, LAN requires the host to have a key, but other LAN players can log in and play the full game.

As for closed-source, open-source markings, may I suggest that we ignore games that are closed-source, but note if they're open sourced? Or maybe put the notice far far away? Otherwise it may make developers feel uncomfortable.

On top of that, what about MMOs like Second Life where the client is open, but the server is not?

**Comment by ssokolow** [comment]
As long as I know where to look so I don't miss the "this game is open-source" flag, I have no problem with that. In fact, given that the vast majority of games are closed-source, that makes more sense from a UI-design perspective anyway.

As for Blockland, definitely tricky to explain clearly. I think the first thing we need to do is figure out where correctness ends and scope-creep begins.

Would it be good enough to just treat key-less operation as a "demo/trial" branch and, therefore, out-of-scope? That would distill the problem back down to "How would one explain Warcraft-style 'you need a key to host a game but not to join one'?"

After all, it's really just a mix of two different well-known patterns:
- Shareware using keys to distribute the trial and full version as the same binary (Which could be handled by listing two branches. One which costs and gives you a key. The other which is free but doesn't.)
- Games allowing a certain number of non-hosting players to free-ride on a hosting player's key.

Second Life is also tricky. I'm probably going to have to find/make time to sit down, diagram this all out, and think on it.

**Comment by ssokolow** [comment]
Wait... I'm obviously not thinking clearly. Second Life is simple to answer. It's just a question of how you define "open-source".

If you want to go for pragmatism, a game like Second Life is open-source for the same reason Desurium is open-source. The code that Desurium installs on the end-user's PC is open, regardless of the status of the server it connects to.

If you want idealism, then it's a three-state flag:
- Closed
- Open Client
- Open System

Yes, I'm sure there are some games out there with open servers and closed clients, but that's not Desura's concern. Desura is a distribution platform for client-side code.

**Comment by ssokolow** [comment]
I've been thinking on Blockland and it might be good enough just build on the suggestion I had before:
- Offer a "full" branch and a "trial" branch with the difference being whether you get a key
- Split "3. LAN play requires CD key uniqueness" into these options:
  - 3. Multiplayer play subject to restrictions (see description)
  - 4. All players must have unique keys or requires all multiplayer games to involve the matchmaking servers
- Explain the Warcraft-style "hosting a game requires a key" restrictions in the game's description.

**Comment by Protektor-Desura** [comment]
I should mention that Second Life is completely open source. You can get the code for the client and the server. The issue there is you can't connect your server to the official servers for people to connect to easily.

I hate getting in to the whole open source source code thing. I think for a large majority of the gamers out there would have no clue what open source means as it relates to their games. They are going to be far more familiar with free or paid.

If we are going to have some tag for open source it should be not obvious since it just confuses users. I also think that the DRM thing should not be obvious since again gamers don't care generally and they aren't all that up on DRM issues. A lot of this is important to open source developers and users but the rest of the world couldn't care less for the most part.

Remember I have to deal with the confused users who don't understand why something is showing up on the system. We also want to make it as easy as possible for users to buy games rather than flood them too much information and overwhelm them to where they say "This is way too complex, let me buy from somewhere else that is simpler."

All of this has to be changed on the server, so you have to convince "Intense!" to make these changes. I personally think if they are added it should be hidden under a pop-up window from a link like "advanced info" or something like that.

**Comment by naryl** [comment]
I think it should be based on what rights do the developers give you.
1. Do (almost) anything you want. (Free software)
2. Run and make copies for personal use (proprietary, no-drm).
3. Only run (drm).

In order to avoid confusing users you can include this information only on the filter/search page. User doesn't really need it when deciding what games to buy unless they're on an anti-drm or anti-proprietary campaign in which case they can check the appropriate boxes on the filter page and forget about them.

Err... does Desura have a filter? :)

**Comment by ssokolow** [comment]
@Protektor-Desura Hide them as deeply as you want, as long as it's less hassle than right-clicking, remembering that Desura's built-in browser has no "copy URL" and no "open in new tab", and then searching up the game in Firefox to check up on things.

@naryl The problem is that, with that system, it doesn't give enough granularity. Either you put something I consider inappropriate under "Run and make copies for personal use" or you include stuff as simple as Minecraft's "must have unique usernames for multiplayer and username can't be set in offline mode" under the blanket heading of "DRM".

**Comment by Protektor-Desura** [comment]
You could create a DRM field that developers can fill out. The biggest problem would come I think if you are trying search/sort by DRM. I personally don't want developers to have to pick from 10+ DRM choices for their games either. The more complex it is the less likely developers want to bother. I think 3-5 generic choices would be about right. I have no idea what the best of them would be but more than 5 might start to confuse/annoy developers. 

I understand some people want to know exactly what kind of DRM a game has, but a majority of gamers just simply don't care other than if it effects their ability to play the game.

**Comment by ssokolow** [comment]
I agree. That's actually part of why I proposed separate entries for single-player and multiplayer functionality. It greatly reduces the combinatorial complexity.

How about this set of questions, which is more geared toward the average player and developer's priorities:
- To access single-player functionality, a user must (choose one):
  1. Just install the game
  2. Key in their unique product key (no network connection required)
  3. Verify their key with the server at install time before they may play
  4. Re-verify their product key with the server periodically to continue playing
- This game supports multiplayer play (yes/no)
  - Has online matchmaking (yes/no)
  - LAN play requires a working connection to the global Internet (yes/no)
  - Multiple players on the same LAN may play using a single purchased copy (yes/no)

The relevance of the multi-player options is pretty obvious, but there is precedent for the single-player one being relevant to the average user too. For example, deployed soldiers, who curse number 4 and sometimes trip over number 3 because of security restrictions.

**Comment by naryl** [comment]
It says nothing about whether you're allowed to develop derived projects.

**Comment by ssokolow** [comment]
Sorry. It's late and I got distracted. I think that, from the average user's perspective, the best way to present open-source would be as the extreme extent of "Does this game have a level editor? Is it moddable?"

Maybe a third drop-down?
- Users may customize their gaming experience by (select most permissive option that applies):
  1. (none)
  2. Loading new resources (levels/maps, player skins, etc.)
  3. Writing and installing mods
  4. Customizing, porting, and modding it (open code, closed resources)
  5. Using it as a starting point for new games (fully open-source)

It may also be possible to present the relevant questions as checkboxes, but I'm starting to crash, so I can't think about the ups and downs of that tonight.

**Comment by Protektor-Desura** [comment]
@naryl I have to be brutally honest and that is the Desura as a platform doesn't care one whit about if you can make derived projects or not. We are in the business of selling games. The question becomes will this get us more users or help us sell more games? I'm not sure one way or the other. If we can come up with something simple then it isn't an issue. 

There is no possible way to cover every possible DRM option out there and I have no interest in trying to create something for that. The reality is there will only be most likely three options for Desura, no DRM, DRM key/serial, DRM auth off Desura (eventually). I can't see us allowing anything else. Right now we are only setup to handle no DRM and DRM key/serial anything else we have had the developer change it. A few developers are asking for DRM Auth so we will support that at some point.

DRM Key/Serial could only be 3 options I think, no network, network once, network every play.
DRM Auth could only be 2 options I think, network every X days, network every play.

The simplest might just be DRM yes/no. It may not tell you want everything you want to know in detail but the developer can always post that in the description of the game if needed.

**Comment by ssokolow** [comment]
@Protektor-Desura Something is always better than nothing and your idea is basically exactly what I'd want from the single-player side of things. 

I'd have to go look up the multiplayer stuff myself, but just being able to easily dismiss anything with DRM auth will probably clear out anything where LAN multiplayer is impossible if the matchmaking servers are taken down.

As long as I clearly understand what I'm paying for, I'll be happy. (I have very strong feelings about any system where, decades down the road, the publisher could go bankrupt and I'd have neither my game nor my money.)

Other information is appreciated, but my primary concern is making sure that I never ever find myself in a situation where a game company effectively tricked me into paying them.

**Comment by Jookia** [comment]
What release should we aim this for?

**Comment by Protektor-Desura** [comment]
This is something that would have to be added server side and would not effect the client since it would just be something listed on the web page for each game. It would have to be added to the database to support it. For it to have any user based on what people here are saying is it would have to have a sort option for displaying games. So again this would have to be put on the list to be eventually/considered to be added to the server side of this. There is no guarantee that Scott (Intense!) would even agree to add something like this.

**Comment by ssokolow** [comment]
For the record, this forum thread has an additional two users who want a UI more conducive to basing purchasing decisions on the presence or absence of DRM:

http://www.desura.com/groups/desura/forum/thread/info-about-drm/

**Comment by ssokolow** [comment]
Well, because of this issue, I just got tricked into paying $5 for a game with online activation so let's hope either Chronic Logic LLC or Desura Pty have the capability to invalidate my CD key and give me a refund or I'm going to have to amend my game-buying policy to limit my Desura games to free stuff and keys from Humble Bundles.

As is, I'm feeling rather betrayed since, no doubt, if I were to open up a PayPal dispute ticket, all you guys would have to do is threaten to close my account, killing updates for all my other games. (This is why I don't use Steam)

**Comment by ssokolow** [comment]
Ok, Scott gave me a refund, so at least the resolution was prompt and fair. I guess I'll just have to do more [than a reasonable amount of] due diligence before jumping on sales.

**Comment by karolherbst** [comment]
If I have time I could look into the information given from the desura server. But I think this should be implemented on the webserver, so it would be displayed in the client, too.

**Comment by ssokolow** [comment]
I recognize that this is more a server-side issue. I've just been drawing a total blank on where I've seen a request for it to be implemented server-side... so I figured, better to say it somewhere than not at all.

**Comment by Mailaender** [comment]
http://www.gamingonlinux.com/articles/desura-to-more-clearly-label-thirdparty-drm.2023
